### PR TITLE
Add sortEndpoints attribute to api-navigation

### PR DESCRIPTION
**Pull request by TwoPlusTwoOne**

### CODE REVIEW

**Comment by jarrodek** [comment]
When we are allowing sorting endpoints we also should be able to sort documentation, types, and security schemes.
Besides, like i commented on the referenced issue, should the API console (and therefore this element) alter the order encoded in the API file? This is not our decision to make as this is someone's design. It may be like this for some reason. I think the navigation should build 2 tree structures for the same endpoint. The spec says that the processors has to respect order.

**Comment by TwoPlusTwoOne** [comment]
> When we are allowing sorting endpoints we also should be able to sort documentation, types, and security schemes.
> Besides, like i commented on the referenced issue, should the API console (and therefore this element) alter the order encoded in the API file? This is not our decision to make as this is someone's design. It may be like this for some reason. I think the navigation should build 2 tree structures for the same endpoint. The spec says that the processors has to respect order.

Normally, I would agree with you, but I've spoken to the people behind the AMF service about the order of the endpoints, and their response was that there is no metadeta stored regarding the order of the endpoints. This means that, when using partial models and using the query service, the endpoints can easily come out of order. And they regularly do.

So what if we exposed the attribute in a way that lets the user know that the API Spec order will 100% not be respected? That way they know the consequences of activating that attribute.

Regarding the other comment, I agree. Adding the option to order documentation, types and security schemes seems like a good idea. I'll add that into the PR ASAP

**Comment by TwoPlusTwoOne** [comment]
I've updated the attribute to be called `arrangeEndpoints` rather than `sortEndpoints`. Setting this attribute will make the navigation arrange the endpoints, maintaining the order of which endpoint appeared first relative to the others.

For example, the following:
```
/transactions/:txId
/billing
/accounts/:accountId
/accounts
/transactions
```

Ends up like this:
```
/transactions
/transactions/:txId
/billing
/accounts
/accounts/:accountId
```

**Comment by jarrodek** [comment]
LGTM
However, I would include a test case where the attribute is actually set, then after model change I would check the arrangement of items in the `_endpoints` array. This tests whether the function is called when the element is constructed.

File: test/api-navigation.test.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ describe('<api-navigation>', () => {
     return (await fixture(`<api-navigation summary selected="test1"></api-navigation>`));
   }
 
+  async function arrangedFixture() {
+    return (await fixture(`<api-navigation rearrangeEndpoints="true"></api-navigation>`));
```
Comment (jarrodek) [inline test/api-navigation.test.js]: Technically HTML attributes are always considered lowercase. This is not a mistake but we should have common coding standards.

File: test/api-navigation.test.js [hunk 2]
```diff
@@ -254,6 +258,48 @@ describe('<api-navigation>', () => {
...
+      { path: "/transactions/:txId" },
+      { path: "/billing" },
+      { path: "/accounts" },
+      { path: "/accounts/:accountId" }
+    ];
+
+    beforeEach(async () => {
+      element = await arrangedFixture();
+      amf = await AmfLoader.load(false, 'rearrange-api');
+      await nextFrame();
```
Comment (jarrodek) [inline test/api-navigation.test.js]: The `nextFrame()` function just schedules a macrotask in the event loop that calls `setTimeout()` function. This is a time for LitElement to commit DOM changes. Here it is not needed as you do not manipulate DOM in tests.

**Pull request merged**